pywbem/mof_compiler.py

```diff
@@ -2178,6 +2191,41 @@ def CreateClass(self, *args, **kwargs):
             self.classes[self.default_namespace] = \
                 NocaseDict({cc.classname: cc})
 
+        objects = list(cc.properties.values())
+        for meth in cc.methods.values():
+            objects += list(meth.parameters.values())
+
+        for obj in objects:
+            # Validate that reference_class exists in repo
+            if obj.type == 'reference':
+                try:
+                    self.GetClass(obj.reference_class, LocalOnly=True,
+                                  IncludeQualifiers=True)
+                except CIMError as ce:
+                    if ce.args[0] == CIM_ERR_NOT_FOUND:
+                        ce.args = (CIM_ERR_INVALID_PARAMETER,
+                                   'Reference object "%s" of class "%s" '
+                                   'reference_class "%s" does not exist' %
+                                   (obj.name, cc.classname,
+                                    obj.reference_class))
+                    raise
```

Comment:
I'm not sure that modifying the existing exception and re-raising it is a good idea. It is more confusing than it helps (the traceback will point to GetClass but the exception text and status code only makes sense for CreateClass). Plus, I seem to recall that `args` was not modifiable in some Python version. But maybe I recall this incorrectly, and its items were not modifiable because it is a tuple.
Anyway, I suggest raising a new exception instead of modifying the existing one. Multiple occurrences of this pattern.

Reply:
OK Will do

Reply:
DONE